x/auth/changepubkey/types/pubkey_history.go

```diff
@@ -0,0 +1,52 @@
+package types
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+func EncodePubKey(cdc codec.BinaryMarshaler, pubkey crypto.PubKey) []byte {
+	if pubkey == nil {
+		return []byte{}
+	}
+	bz, err := codec.MarshalAny(cdc, pubkey)
+	if err != nil {
+		panic(err)
+	}
+	return bz
+}
+
+func DecodePubKey(cdc codec.BinaryMarshaler, bz []byte) crypto.PubKey {
+	if len(bz) == 0 {
+		return nil
+	}
+	var pubkey crypto.PubKey
+	err := codec.UnmarshalAny(cdc, &pubkey, bz)
+	if err != nil {
+		panic(err)
+	}
+	return pubkey
+}
+
+func EncodeHistoricalEntry(entry PubKeyHistory) []byte {
+	// TODO: what steps are required to use MarshalAny for PubKeyHistory
```

Comment:
You're going to need to either pass in the cdc into this function, or move this function into the keeper.  I think the first makes more sense.

Reply:
Also, why do we need to use MarshalAny? Isn't that for interfaces?

Reply:
I fixed it to use mustmarshalbinanrybare instead now
https://github.com/sikkatech/cosmos-sdk/pull/61/commits/d4a0ab401396571649dd0d35670e2bf459e84df5